src/client.ts

```diff
@@ -1,3 +1,81 @@
+import { PlayerInfo, SessionInfo } from "./types";
+import axios from 'axios'
+
 export default class ArcadeClientSDK {
-  
+
+  sessionToken: string = ""
+  url = "https://userapi.ultimatearcade.io"
+
+  constructor(options?: {
+    baseDomain?: string
+  }) {
+    // Register listener for iframe messaging
+    window.onmessage = (e) => {
```

Comment:
The sender can't know if its message got received. We therefore should also add some kind of "ready" message back to the parent.